storage/watchable_store.go

```diff
@@ -448,13 +381,99 @@ type watcher struct {
 	ch chan<- WatchResponse
 }
 
-// unsafeAddWatcher puts watcher with key k into watchableStore's synced.
-// Make sure to this is thread-safe using mutex before and after.
-func unsafeAddWatcher(synced *map[string]map[*watcher]struct{}, k string, wa *watcher) error {
+// kvsFromUnsynced returns all key bytes and raw storagepb.KeyValue bytes
+// for all unsynced watchers. And returns prefixes, a map of prefixes that
+// need to be watched. store must be mutex locked before this function.
+func kvsFromUnsynced(s *watchableStore) ([][]byte, [][]byte, map[string]struct{}) {
```

Comment:
The returned 3-tuple would be easier to work with here and elsewhere as a struct (`PrefixedKVs`? `WatchKVs`?)